openshift/templates/iiw-book-agent/iiw-book-agent-deploy.yaml

```diff
@@ -4,48 +4,102 @@ apiVersion: v1
+  #       app: ${APP_GROUP}
+  #       app-group: ${APP_GROUP}
+  #       env: ${TAG_NAME}
+  #   spec:
+  #     description: |
+  #       Define the network parameters for accessing remote agents.
+  #     entries:
+  #       - 0.0.0.0/0
+  #     servicePorts:
+  #       - tcp/9700:9799
+  #       - tcp/80
+  #       - tcp/443
+  # - kind: NetworkSecurityPolicy
+  #   apiVersion: security.devops.gov.bc.ca/v1alpha1
+  #   metadata:
+  #     name: ${NAME}
+  #     labels:
+  #       name: ${NAME}
+  #       env: ${TAG_NAME}
+  #       app: ${APP_GROUP}
+  #       app-group: ${APP_GROUP}
+  #   spec:
+  #     description: Allow the the agent to access the service and external network
+  #     source:
+  #       - - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+  #         - role=${ROLE}
+  #         - app=${APP_GROUP}
+  #         - env=${TAG_NAME}
+  #     destination:
+  #       - - ext:name=${NAME}
+  #       - - role=service
+  #         - app=${APP_GROUP}
+  #         - env=${TAG_NAME}
+  #         - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+  - kind: NetworkPolicy
+    apiVersion: networking.k8s.io/v1
     metadata:
-      name: ${NAME}
-      network: ${NAME}
+      name: ${NAME}-allow-ingress
       labels:
-        name: ${NAME}
-        network: ${NAME}
+        name: ${NAME}-allow-ingress
         app: ${APP_GROUP}
         app-group: ${APP_GROUP}
+        role: ${ROLE}
         env: ${TAG_NAME}
     spec:
       description: |
-        Define the network parameters for accessing remote agents.
-      entries:
-        - 0.0.0.0/0
-      servicePorts:
-        - tcp/9700:9799
-        - tcp/80
-        - tcp/443
-  - kind: NetworkSecurityPolicy
-    apiVersion: security.devops.gov.bc.ca/v1alpha1
+        Allow traffic from routes into the ${ROLE} container(s).
+      ingress:
+        - from:
+            - namespaceSelector:
+                matchLabels:
+                  network.openshift.io/policy-group: ingress
+      podSelector:
+        matchLabels:
+          role: ${ROLE}
+          app: ${APP_GROUP}
+          env: ${TAG_NAME}
+      policyTypes:
+        - Ingress
+  - kind: NetworkPolicy
+    apiVersion: networking.k8s.io/v1
     metadata:
       name: ${NAME}
       labels:
         name: ${NAME}
-        env: ${TAG_NAME}
         app: ${APP_GROUP}
-        app-group: ${APP_GROUP}
+        env: ${TAG_NAME}
     spec:
-      description: Allow the the agent to access the service and external network
-      source:
-        - - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
-          - role=${ROLE}
-          - app=${APP_GROUP}
-          - env=${TAG_NAME}
-      destination:
-        - - ext:name=${NAME}
-        - - role=service
-          - app=${APP_GROUP}
-          - env=${TAG_NAME}
-          - $namespace=${NAMESPACE_NAME}-${TAG_NAME}
+      description: Allow the service to access the agent.
+      ingress:
+        - from:
+            - podSelector:
+                matchLabels:
+                  role: service
```

Comment:
The role of the service can be changed at deployment time by updating the parameters: this should also be populated by a parameter, in order to make it possible to update it if necessary.

Reply:
change completed